proxy/src/registry.rs

```diff
@@ -118,19 +141,22 @@ impl Registry {
         );
         // TODO(xla): Unpack the result to find out if the application of the transaction failed.
         let register_applied = self.client.submit_transaction(register_tx).await?.await?;
-
-        Ok(Transaction {
+        let tx = Transaction {
             id: register_applied.tx_hash,
             messages: vec![Message::OrgRegistration(org_id)],
             state: TransactionState::Applied(register_applied.block),
             timestamp: SystemTime::now(),
-        })
+        };
+
+        self.transactions.insert(tx.id, tx.clone());
```

Comment:
Might be a good idea to implement a `self.submit_tx` proxy function that calls `self.client.submit_transaction` and inserts the tx in the cache. That way we don't ever forget to cache stuff.

Reply:
Unfortunately that is not doable at the moment as `Message` in the registry is a trait with a static method. So we can't really pull the code out that constructs registry transactions.